src/Sparrow/Threading/DisposeOnce.cs

```diff
@@ -102,10 +144,10 @@ public bool Disposed
                 if (state.Item1 == false)
                     return false;
 
-                if (typeof(TOperationMode) == typeof(SingleAttempt))
+                if (typeof(TOperationMode) == typeof(SingleAttempt) && _operationModeData.DuringDispose == false)
                     return true;
```

Comment:
Better to change to: `return  _operationModeData.DuringDispose == false;`, no?

Reply:
right